src/test/java/com/google/cloud/tools/appengine/cloudsdk/serialization/CloudSdkComponentTest.java

```diff
@@ -0,0 +1,154 @@
+public class CloudSdkComponentTest {
+
+  private Gson gson;
+
+  @Before
+  public void setup() {
+    this.gson = new Gson();
+  }
+
+  @Test
+  public void testGsonDeserializeJson() {
+    CloudSdkComponent result = gson.fromJson(getJson(), CloudSdkComponent.class);
+    CloudSdkComponent expected = getCloudSdkComponent();
+    assertCloudSdkComponentsEqual(expected, result);
+  }
+
+  @Test
+  public void testGsonSerialization() {
+    CloudSdkComponent cloudSdkComponent = getCloudSdkComponent();
+    String result = gson.toJson(cloudSdkComponent);
+
+    // Since the ordering of fields in JSON objects is not guaranteed, we cannot compare the full
+    // strings for equality. Instead, use regexes to validate that key/value pairs are present.
+    assertJsonKeyValueExists("current_version_string", cloudSdkComponent.getCurrentVersion(),
+        result);
+    assertJsonKeyValueExists("id", cloudSdkComponent.getId(), result);
+    assertJsonKeyValueExists("is_configuration", cloudSdkComponent.getIsConfiguration(), result);
+    assertJsonKeyValueExists("is_hidden", cloudSdkComponent.getIsHidden(), result);
+    assertJsonKeyValueExists("latest_version_string", cloudSdkComponent.getLatestVersion(),
+        result);
+    assertJsonKeyValueExists("name", cloudSdkComponent.getName(), result);
+    assertJsonKeyValueExists("size", cloudSdkComponent.getSizeInBytes(), result);
+  }
+
+  @Test
+  public void testGsonSerializationDeserialization() {
+    CloudSdkComponent initial = getCloudSdkComponent();
+    String serialized = gson.toJson(initial);
+    CloudSdkComponent result = gson.fromJson(serialized, CloudSdkComponent.class);
+    assertCloudSdkComponentsEqual(initial, result);
+  }
+
+  @Test
+  public void testStateIsInstalled_true() {
+    CloudSdkComponent.State state = new CloudSdkComponent.State();
+    state.setName("Installed");
+    assertTrue(state.isInstalled());
+  }
+
+  @Test
+  public void testStateIsInstalled_false() {
+    List<String> notInstalledStates = Arrays.asList("uninstalled", "not installed", "", null);
+    for (String stateName : notInstalledStates) {
+      CloudSdkComponent.State state = new CloudSdkComponent.State();
+      state.setName(stateName);
+      assertFalse(state.isInstalled());
+    }
+  }
+
+  private void assertJsonKeyValueExists(String expectedKey, String expectedValue, String result) {
+    String regex = String.format(".*%s\":\\s*\"%s\".*", expectedKey, expectedValue);
+    assertTrue(result.matches(regex));
+  }
+
+  private void assertJsonKeyValueExists(String expectedKey, int expectedValue, String result) {
+    String regex = String.format(".*%s\":\\s*%s.*", expectedKey, expectedValue);
+    assertTrue(result.matches(regex));
+  }
+
+  private void assertJsonKeyValueExists(String expectedKey, boolean expectedValue, String result) {
+    String regex = String.format(".*%s\":\\s*%s.*", expectedKey, expectedValue);
+    assertTrue(result.matches(regex));
+  }
+
+  private String getJson() {
```

Comment:
This is the expected json-ified version of the CloudSdkComponent test fixture in this class. To clarify, I'm updating these two method names to `getCloudSdkComponentTestFixture` and `getCloudSdkComponentTestFixtureAsJson`.